src/js/app.js

```diff
@@ -0,0 +1,196 @@
+     * @param {String} dom element to get
+     * @return {String} selected value
+     */
+    $scope.selected = (id) => {
+      return document.getElementById(id || 'indexed-files').value;
+    };
+
+
+    /**
+     * @param  {String} fileName - file name
+     * @param  {Object} data  - data to transform
+     * @param  {count} count - count of object in the file
+     */
+    $scope.transformData = (fileName, data, count) => {
+      return {
+        fileName: fileName,
+        data: data,
+        count: count
+      };
+    };
+
+    /**
+     * creates an inverted Index 
+     */
+    $scope.createIndex = () => {
+      const selected = $scope.selected('uploaded-files');
+      const uploadedFiles = $scope.uploadedFiles;
+      console.log(selected);
+      if ($scope.isValidFile(selected) &&
+        uploadedFiles.hasOwnProperty(selected)) {
+        const data = indexer.createIndex(selected, uploadedFiles[selected]);
+        if (data === 'error' || $scope.length(data) < 1) {
+          alerts('invalid json file', 'danger');
+          return false;
+        }
+        const fileCount = uploadedFiles[selected].length;
+        $scope.index[0] = $scope.transformData(selected, data, fileCount);
+        $scope.fileCount[selected] = fileCount;
+        $scope.allIndexedFiles.push(selected);
+        $scope.showTable = true;
+      }
+    };
+
+    /** searches for a query
+     *
+     * @param {String} query = value to be searched
+     * @return {void}  set the index value
+     */
+    $scope.searchIndex = (query) => {
+      let result;
+      const selected = document.getElementById('indexed-files').value;
+      let fileCount = null;
+      $scope.index = [];
+      // checks if a query was passed in
+      if (!query || selected === "--select a file--") {
+        alerts("please enter a query and select file  to search",
+          "danger ", true, 5000);
+        return false;
+      }
+      // displays serach result for all indexed files
+      if (selected === 'all') {
+        let count = 0;
+        for (let file in indexer.indexedFiles) {
+          let searchData = indexer.searchIndex(query, file);
+          fileCount = $scope.fileCount[file];
+          if ($scope.length(searchData) < 1) {
+            continue;
+          }
+          $scope.index[count] = $scope
+            .transformData(file, searchData, fileCount);
+          count++;
+        }
+        if ($scope.index.length < 1) {
+          alert('word does not exist in any file', 'danger');
+        }
+      } else {
+        fileCount = $scope.fileCount[selected];
+        result = indexer.searchIndex(query, selected);
+        $scope.index[0] = $scope.transformData(selected, result, fileCount);
+      }
+    };
+
+
+    /** checks the value to be selected
+     *
+     * @param {String}  value to check
+     * @return {Boolen}  true or flase
+     */
+    $scope.isValidFile = function (file) {
+      return file.match(/\.json$/);
```

Comment:
You can use HTML5 accept attribute to validate the file that type that can be uploaded and just leave the remaining validation to the `InvertedIndex` Class